src/pages/workspaces/workspace/applications/AppLauncher.js

```diff
@@ -15,13 +15,27 @@ import * as Utils from 'src/libs/utils'
 import { wrapWorkspace } from 'src/pages/workspaces/workspace/WorkspaceContainer'
 
 
+const getSparkInterfaceSource = (proxyUrl, sparkInterface) => {
+  console.assert(_.endsWith('/jupyter', proxyUrl), 'Unexpected ending for proxy URL')
+  const proxyUrlWithlastSegmentDropped = _.flow(_.split('/'), _.dropRight(1), _.join('/'))(proxyUrl)
+  return `${proxyUrlWithlastSegmentDropped}/${sparkInterface}`
+}
+
+const getApplicationIFrameSource = (proxyUrl, application, sparkInterface) => {
+  return Utils.switchCase(application,
+    ['terminal', () => `${proxyUrl}/terminals/1`],
+    ['spark', () => getSparkInterfaceSource(proxyUrl, sparkInterface)],
+    [Utils.DEFAULT, () => proxyUrl] // RStudio
+  )
+}
+
 const ApplicationLauncher = _.flow(
   Utils.forwardRefWithName('ApplicationLauncher'),
   wrapWorkspace({
     breadcrumbs: props => breadcrumbs.commonPaths.workspaceDashboard(props),
     title: _.get('application')
   })
-)(({ namespace, name, refreshRuntimes, runtimes, persistentDisks, application, workspace }, ref) => {
+)(({ sparkInterface, refreshRuntimes, runtimes, persistentDisks, application, workspace }, ref) => {
   const cookieReady = Utils.useStore(cookieReadyStore)
   const [showCreate, setShowCreate] = useState(false)
```

Comment:
huh... not sure why but the compute modal pops up in my face briefly when I navigate from a different page to the appLauncher sometimes...

It does it in dev too so not from this PR... might be worth a bug ticket (would be good to not hold this up)

Reply:
Ticket: https://broadworkbench.atlassian.net/browse/IA-2985